udata_front/theme/gouvfr/assets/js/composables/useDropdown.ts

```diff
@@ -0,0 +1,87 @@
+import { ref, onMounted, onUnmounted, MaybeRefOrGetter, toValue } from 'vue';
+import { useCollapse } from './useCollapse';
+import useActiveDescendant from './useActiveDescendant';
+
+type Option = {
+  id: string;
+  [key: string]: string | number | boolean | undefined;
+}
+
+export function useDropdown(options: MaybeRefOrGetter<Option[]>, q: MaybeRefOrGetter<string>) {
```

Comment:
We can use a conditional type, instead of defining our custom Option because we are only interested in the `id`

Reply:
I forgot to add the links : https://www.typescriptlang.org/docs/handbook/2/conditional-types.html and https://www.typescriptlang.org/docs/handbook/2/objects.html#extending-types